services/src/main/java/org/fao/geonet/api/GlobalExceptionController.java

```diff
@@ -148,15 +149,29 @@ public Object securityHandler(final HttpServletRequest request, final Exception
     }
 
     @ResponseBody
-    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ResponseStatus(HttpStatus.PAYLOAD_TOO_LARGE)
     @ApiResponse(content = {@Content(mediaType = APPLICATION_JSON_VALUE)})
     @ExceptionHandler({
         MaxUploadSizeExceededException.class
     })
-    public ApiError maxFileExceededHandler(final Exception exception) {
-        storeApiErrorCause(exception);
+    public ApiError maxFileExceededHandler(final Exception exception, final HttpServletRequest request) {
+        Exception ex;
+        long contentLength = request.getContentLengthLong();
+        // As MaxUploadSizeExceededException is a sping exception, we need to convert it to a localized exception so that it can be translated.
```

Comment:
```suggestion
        // As MaxUploadSizeExceededException is a Spring exception, we need to convert it to a localized exception so that it can be translated.
```

Reply:
Fixed